### ISSUE TITLE
River - Input Entity + Input Not Powered Trigger - False Alerts?

### ISSUE THREAD

**Issue by Jiriteach** [vwt12eh8/hassio-ecoflow#29]
Hi - I am using the latest version 2.2.3 and still seeing false alerts with the input entity and the trigger for input not powered. See the screenshot.

This device is on bypass and I know input power was not lost so not sure why I am seeing these? Any ideas?

Thanks

![Screen Shot 2022-08-03 at 12 10 21](https://user-images.githubusercontent.com/2116446/182497666-aa11e824-7252-4c89-937f-60619884b9b0.png)

**Comment by vwt12eh8** [comment]
`became unavailable` means that the connection to the device has been lost.
Unlike `undetected`, it does not mean that the input is turned off.

I'm sure I've fixed the problem with the connection becoming unavailable before, but if it's recurring, I'll need to investigate...

**Comment by Jiriteach** [comment]
Thanks and good to know. I monitor uptime on the River's via my monitoring tools and havent noticed any drops but I will keep an eye on this.

The trigger for input not powered shouldnt fire for become unavailable though right?

**Comment by vwt12eh8** [comment]
Yes, When there is really no input, the state should be `off`, not `unavailable`.

**Comment by Jiriteach** [comment]
Ok - after some testing I can confirm that the trigger for `input not powered` is firing if network connectivity drops (ie. the River drops off wifi).

This should not be happenning as its false alert since actual power has not dropped. So looks like a bug. Thanks

**Comment by vwt12eh8** [comment]
Are you sure you didn't set the warning condition to `state ! = 'on'`?
To make it work correctly, you need to set either `state == 'off'` or `state ! = 'on' and state ! = 'unavailable'`.

**Comment by Jiriteach** [comment]
> Are you sure you didn't set the warning condition to `state ! = 'on'`?
> 
> To make it work correctly, you need to set either `state == 'off'` or `state ! = 'on' and state ! = 'unavailable'`.

Am using the provided trigger as follows. Perhaps I should set a duration in there.

![image](https://user-images.githubusercontent.com/2116446/183233346-29d0e1d4-781a-414b-8b12-741e43703365.png)

**Comment by vwt12eh8** [comment]
Perhaps that device trigger internally uses `state ! = 'on'` and not expecting `unavailable`.

It's a bit more challenging, but I think you can ignore `unavailable` by setting the trigger type to State.

![image](https://user-images.githubusercontent.com/63479906/183233792-fbd31a75-c9c3-41ca-bf35-f63070cf46b5.png)

```yml
trigger:
  - platform: state
    entity_id:
      - binary_sensor.river_******_input
    to: "off"
```

**Comment by vwt12eh8** [comment]
Note that a device is considered disconnected when it has not sent any data for 15 consecutive seconds.

**Comment by Jiriteach** [comment]
> Note that a device is considered disconnected when it has not sent any data for 15 consecutive seconds.

Good to know! Have set a duration longer than 30 sec so it will wait  so should work well. Thanks

**Comment by Jiriteach** [comment]
So even with the delays, still getting lots of alerts and pretty confident the devices are not disconnecting from wifi so - I have now changed my automations to use a state trigger as you have suggested and will see how this goes.
